include/bout_types.hxx

```diff
@@ -65,13 +65,17 @@ const std::string& DIRECTION_STRING(DIRECTION direction);
 ///   coordinates
 enum class YDirectionType { Standard, Aligned };
 
+const std::string& convertYDirectionTypeToString(YDirectionType d);
```

Comment:
I know this just following the existing naming convention for enum -> string conversion functions in this header, but we do have `toString(T)` in [utils.hxx](https://github.com/boutproject/BOUT-dev/blob/master/include/utils.hxx#L442). I think it would be nice if we just overloaded `toString` for all the enums.

I'll make a PR now for the others

Reply:
I'll update this PR to follow the style of your PR when you're done. Trying to dumbly replace `convertYDirectionTypeToString` with `toString` failed because we can't include `utils.hxx` in `bout_types.hxx`...